### PR TITLE
updating to allow conditional creation of the cdn and dns

### PR DESCRIPTION
**Pull request by dannybrody**
Currently we require the ability to conditionally create the aws_cloudfront_distribution.  Updated to allow the conditional creation of resource "aws_cloudfront_distribution" "default" together with module "dns".  Also updated the outputs to work with both cases.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -118,6 +119,6 @@ module "dns" {
   aliases          = "${var.aliases}"
   parent_zone_id   = "${var.parent_zone_id}"
   parent_zone_name = "${var.parent_zone_name}"
-  target_dns_name  = "${aws_cloudfront_distribution.default.domain_name}"
-  target_zone_id   = "${aws_cloudfront_distribution.default.hosted_zone_id}"
+  target_dns_name  = "${element(concat(aws_cloudfront_distribution.default.*.domain_name, list("")), 0)}"
```
Comment (aknysh) [inline main.tf]: More readable syntax:

```
"${join("", aws_cloudfront_distribution.default.*.domain_name)}"
```

we use the `splat+join` pattern everywhere.
`join` will return an empty string if the list does not contain any items.

File: outputs.tf [hunk 2]
```diff
@@ -1,11 +1,11 @@
 output "cf_id" {
-  value       = "${aws_cloudfront_distribution.default.id}"
+  value       = "${element(concat(aws_cloudfront_distribution.default.*.id, list("")), 0)}"
```
Comment (aknysh) [inline outputs.tf]: use `splat+join`

File: outputs.tf [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
   description = "ID of AWS CloudFront distribution"
 }
 
 output "cf_arn" {
-  value       = "${aws_cloudfront_distribution.default.arn}"
-  description = "ID of AWS CloudFront distribution"
+  value       = "${element(concat(aws_cloudfront_distribution.default.*.arn, list("")), 0)}"
```
Comment (aknysh) [inline outputs.tf]: use `splat+join`

File: outputs.tf [hunk 4]
```diff
@@ -14,22 +14,22 @@ output "cf_aliases" {
 }
 
 output "cf_status" {
-  value       = "${aws_cloudfront_distribution.default.status}"
+  value       = "${element(concat(aws_cloudfront_distribution.default.*.status, list("")), 0)}"
```
Comment (aknysh) [inline outputs.tf]: use `splat+join`

File: outputs.tf [hunk 5]
```diff
@@ -14,22 +14,22 @@ output "cf_aliases" {
...
   description = "Current status of the distribution"
 }
 
 output "cf_domain_name" {
-  value       = "${aws_cloudfront_distribution.default.domain_name}"
+  value       = "${element(concat(aws_cloudfront_distribution.default.*.domain_name, list("")), 0)}"
```
Comment (aknysh) [inline outputs.tf]: use `splat+join`

File: outputs.tf [hunk 6]
```diff
@@ -14,22 +14,22 @@ output "cf_aliases" {
...
   description = "Domain name corresponding to the distribution"
 }
 
 output "cf_etag" {
-  value       = "${aws_cloudfront_distribution.default.etag}"
+  value       = "${element(concat(aws_cloudfront_distribution.default.*.etag, list("")), 0)}"
```
Comment (aknysh) [inline outputs.tf]: use `splat+join`

File: outputs.tf [hunk 7]
```diff
@@ -14,22 +14,22 @@ output "cf_aliases" {
...
   description = "Current version of the distribution's information"
 }
 
 output "cf_hosted_zone_id" {
-  value       = "${aws_cloudfront_distribution.default.hosted_zone_id}"
+  value       = "${element(concat(aws_cloudfront_distribution.default.*.hosted_zone_id, list("")), 0)}"
```
Comment (aknysh) [inline outputs.tf]: use `splat+join`

**Comment by dannybrody** [comment]
@aknysh sounds great, I'll update, test, and then update the PR.  Thanks for responding so quickly!

**Comment by dannybrody** [comment]
@aknysh I went ahead and tried updated the other modules with `count = var.enabled == "true" ? 1 : 0` however, it tried to destroy my resources.  when i updated it to `enabled    = "${var.enabled ? true : false }"` it worked as expected

**Comment by dannybrody** [comment]
Any updates on this?

**Comment by dannybrody** [comment]
updated due to running into the following plan error:
`Error: module.application_cloudfront.aws_cloudfront_distribution.default: restrictions.0.geo_restriction.0.locations: should be a list`

File: main.tf [hunk 8]
```diff
@@ -1,5 +1,6 @@
 module "origin_label" {
   source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.3.7"
+  enabled    = "${var.enabled ? true : false }"
```
Comment (aknysh) [inline main.tf]:
```suggestion
  enabled    = "${var.enabled == "true" ? true : false }"
```

File: main.tf [hunk 9]
```diff
@@ -13,7 +14,8 @@ resource "aws_cloudfront_origin_access_identity" "default" {
 }
 
 module "logs" {
-  source                   = "git::https://github.com/cloudposse/terraform-aws-log-storage.git?ref=tags/0.2.2"
+  source                   = "git::https://github.com/cloudposse/terraform-aws-log-storage.git?ref=tags/0.3.0"
+  enabled                  = "${var.enabled ? true : false }"
```
Comment (aknysh) [inline main.tf]:
```suggestion
  enabled                  = "${var.enabled}"
```

File: main.tf [hunk 10]
```diff
@@ -28,6 +30,7 @@ module "logs" {
 
 module "distribution_label" {
   source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.3.7"
+  enabled    = "${var.enabled ? true : false }"
```
Comment (aknysh) [inline main.tf]:
```suggestion
  enabled    = "${var.enabled}"
```

File: main.tf [hunk 11]
```diff
@@ -37,6 +40,7 @@ module "distribution_label" {
 }
 
 resource "aws_cloudfront_distribution" "default" {
+  count               = "${var.enabled ? 1 : 0 }"
```
Comment (aknysh) [inline main.tf]:
```suggestion
  count               = "${var.enabled == "true" ? 1 : 0 }"
```

**Pull request closed without merging**